analyzers/src/SonarAnalyzer.CSharp/Rules/RedundancyInConstructorDestructorDeclaration.cs

```diff
@@ -106,5 +127,9 @@ private static bool TypeHasExactlyOneConstructor(BaseMethodDeclarationSyntax con
 
         private static bool IsConstructorParameterless(BaseMethodDeclarationSyntax constructorDeclaration) =>
             constructorDeclaration.ParameterList.Parameters.Count == 0;
+
+        private static bool ContainsInitializedFieldOrProperty(INamedTypeSymbol symbol) =>
+            symbol.GetMembers().OfType<IFieldSymbol>().Any(f => f.DeclaringSyntaxReferences.Any(d => ((VariableDeclaratorSyntax)d.GetSyntax()).Initializer != null))
```

Comment:
If `f` cannot be `x`, because also `p` would need to be `x`, I would give more meaningful names: `property` and `field`.
`!= null` should be `is not null`